core/domain/feedback_services.py

```diff
@@ -366,6 +366,30 @@ def _get_thread_from_model(thread_model):
         thread_model.created_on, thread_model.last_updated)
 
 
+def get_new_messages_for_exp_id(exp_id):
+    """Fetch the feedback threads for the given exploration, and use those to
+    finally get the new (open) feedback messages.
+
+    Args:
+        exp_id: str.
+
+    Returns:
+       A list of FeedbackMessage.
+    """
+    thread_models = (
+        feedback_models.FeedbackThreadModel.get_latest_updated_threads(exp_id))
+
+    message_models = []
+    for thread_model in thread_models:
+        message_models.append(
+            feedback_models.FeedbackMessageModel.get_most_recent_message(
```

Comment:
I don't think so - since a get_multi version would actually be passed in a list of thread ids for which we want the most recent message out of, and it'll fetch the most recent messages for which `cls.thread_id.IN(thread_ids_list)` followed by `.order(-cls.last_updated)`. This resultant list would be mostly larger in size (since it'll not be ensured that there is just a single message out of each thread) and thus could might even take more time than 3 simple get calls!
or am I missing something ?